src/imap/idle.rs

```diff
@@ -18,6 +18,9 @@ pub enum Error {
     #[fail(display = "IMAP IDLE protocol failed to init/complete")]
     IdleProtocolFailed(#[cause] async_imap::error::Error),
 
+    #[fail(display = "IMAP IDLE protocol timed out")]
+    IdleTimeout(String),
```

Comment:
I would add a proper cause here, wrapping actual `async_imap` error instead of its string version.